ipn/backend.go

```diff
@@ -103,6 +103,9 @@ type Options struct {
 	// TODO(danderson): remove some time after the transition to
 	// tailscaled is done.
 	LegacyConfigPath string
+	// Hostname is the hostname to use for identifying the node. If
+	// not set, os.Hostname is used.
+	Hostname string
```

Comment:
Any way to do this in ipn.Prefs and SetPrefs instead? The Start call is a mess of "load this state but then also apply these random other config tweaks", and I'd prefer to not make it worse. In particular, things configured in Start don't get persisted in the ipn state store.